### PR TITLE
chore: Optimize CI workflow with slim builds

### PR DESCRIPTION
**Pull request by binarylogic**
This change optimizes our CI workflow to take advantage of settings that produce slim builds. This is required because the CI machines only have 12gb of disk space, something we were routinely running out of. This should make CI and it's tests much more stable.

Supercedes https://github.com/timberio/vector/pull/2383

### CODE REVIEW

**Comment by binarylogic** [comment]
@Hoverbear how do you feel about this approach over templatizing our CI workflows? This also represents the initial steps to clean up our Makefile, which I plan to follow up with.

Finally, notice my last commit that disabled the Cargo changes that produce slim builds. Removing those settings make the tests pass consistently, adding them result in one test failing over and over.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -134,13 +124,12 @@ jobs:
...
-      FEATURES: cloudwatch-logs-integration-tests,cloudwatch-metrics-integration-tests,ec2-metadata-integration-tests,firehose-integration-tests,kinesis-integration-tests,s3-integration-tests
     steps:
       - uses: actions/checkout@v1
-      - run: docker-compose up -d localstack mockwatchlogs ec2_metadata minio
-      - run: cargo test --no-default-features --features $FEATURES
+      - run: make slim-builds
+      - run: make test-integration-docker
 
-  docker-integration-tests:
+  test-integration-elasticsearch:
```
Comment (Hoverbear) [inline .github/workflows/ci.yml:132]: I do kind of think keeping the templating from #2383 was useful here and for future work.

**Comment by Hoverbear** [comment]
Hey if you can get them to pass without it, awesome!

**Comment by Hoverbear** [comment]
This looks fine, I do really want to eventually introduce templating or at least anchor support for our workflows later though.

**Comment by binarylogic** [comment]
I do too. Let's see how long we can kick the can before doing so. I imagine GHA has acnhors on their roadmap.

**Pull request merged**